### PR TITLE
Add regression based technique for calibrating k-mer counts

### LINKED ISSUE #54: Calibrate k-mer counts vs. sequence context
Sequencing is preferentially biased towards `GC` rich sequences. We should correct for this by regressing the _k_-mer counts versus sequence context.

### PR DESCRIPTION
**Pull request by fnothaft**
Closes #54. Translates _k_-mers into dinucleotide based feature vectors, and then trains a linear regression model on these feature vectors. The mean coverage and the residual for this _k_-mer are used to remove bias.

### CODE REVIEW

**Comment by tdanford** [comment]
Frank, can you rebase again?

**Comment by fnothaft** [comment]
Sure; give me a min, need to resolve non-trivial merge conflicts...

**Comment by tdanford** [comment]
No worries.  Take your time, ping me on gitter when you're ready.

File: RNAdam-core/src/main/scala/org/bdgenomics/RNAdam/algorithms/quantification/Tare.scala [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+
+  /**
+   * Converts a base into an integer number. Only works for A/C/G/T, both
+   * lower and upper case. Will throw an exception for other bases, user is
+   * expected to check before passing to function.
+   *
+   * @param base The base to convert into an integer.
+   * @return Returns a mapping from a base to an integer.
+   */
+  private def idx(base: Char): Int = base match {
```
Comment (tdanford) [inline RNAdam-core/src/main/scala/org/bdgenomics/RNAdam/algorithms/quantification/Tare.scala]: I'm not saying there's a change to be made here now, but this is yet another use-case for having a separate 'alphabet' style value, with encodings and conversions for individual letters.

**Comment by tdanford** [comment]
LGTM -- thanks @fnothaft !

**Pull request merged**